app.rb

```diff
@@ -6,20 +6,40 @@
 require_relative './lib/players/human.rb'
 require_relative './lib/ui/user_interface.rb'
 require_relative './lib/ui/io.rb'
+require_relative './lib/scripts/setup_script.rb'
+require_relative './lib/validators/setup_validator.rb'
 
-def initializer(size, token1, token2)
+def initializer(size)
+  player1 = :'Player 1'
+  player2 = :'Player 2'
   board = Board.new(size, '-')
   checker = BoardChecker.new(3, '-')
   io = IO.new(STDIN, STDOUT)
-  user_interface = UserInterface.new(io, board, token1, token2)
+  user_interface = UserInterface.new(io, board)
+  setup_script = SetupScript.new
+  setup_validator = SetupValidator.new(setup_script)
+
+  user_interface.welcome
+  token1 = select_token(user_interface, setup_script, setup_validator, player1, '')
+  token2 = select_token(user_interface, setup_script, setup_validator, player2, token1)
+
   human = Human.new(board, user_interface, token1)
   ai = AI.new(checker, token2, token1)
   player1 = Player.new(human, token1, user_interface)
   player2 = Player.new(ai, token2, user_interface)
   game = Game.new(board, checker, player1, player2)
 
-  user_interface.welcome
   game.play
 end
 
-initializer(3, 'X', 'O')
+def select_token(user_interface, script, setup_validator, player, compare_to)
```

Comment:
could we make generating `"\n#{player}#{script.token_selection}"` a method on the script instead?

Reply:
Added this to the script.